src/Generator/Driver.cs

```diff
@@ -316,14 +316,35 @@ public void SaveCode(IEnumerable<GeneratorOutput> outputs)
                     var fileRelativePath = $"{fileBase}.{template.FileExtension}";
 
                     var file = Path.Combine(outputPath, fileRelativePath);
-                    File.WriteAllText(file, template.Generate());
+                    WriteGeneratedCodeToFile(file, template.Generate());
                     output.TranslationUnit.Module.CodeFiles.Add(file);
 
                     Diagnostics.Message("Generated '{0}'", fileRelativePath);
                 }
             }
         }
 
+        private void WriteGeneratedCodeToFile(string file, string generatedCode)
+        {
+            var fi = new FileInfo(file);
+
+            // We only compare contents if the generated code length is the same as the length of the existing file. If their lengths are different then they are
+            // definitely no the same so we should overwrite.
+            if (fi.Exists && fi.Length == generatedCode.Length)
```

Comment:
Would you mind rewriting as a single statement, like this:
```C#
if (!fi.Exists || fi.Length != generatedCode.Length ||
    File.ReadAllText(file) != generatedCode)
    File.WriteAllText(file, generatedCode);
```

Reply:
Done